src/app/tools/point_shapes.h

```diff
@@ -65,7 +70,7 @@ class BrushPointShape : public PointShape {
   bool m_firstPoint;
   Brush* m_lastBrush;
   BrushType m_origBrushType;
-  std::array<std::shared_ptr<CompressedImage>, 4> m_compressedImages;
+  std::array<std::shared_ptr<CompressedImage>, 8> m_compressedImages;
```

Comment:
Change the 8 with the `BrushIndex::ELEMENTS` constant:
```suggestion
  std::array<std::shared_ptr<CompressedImage>, int(BrushIndex::ELEMENTS)> m_compressedImages;
```